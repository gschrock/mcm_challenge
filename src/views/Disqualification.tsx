import React from "react";
import { useHistory } from "react-router-dom";

interface Props {
  loanResponse: { [key: string]: string | boolean } | null;
}

export const Disqualification: React.SFC<Props> = ({ loanResponse }) => {
  let history = useHistory();
  /**
   * If user attempts to navigate here without loan
   * response data redirect them back to landing route.
   */
  if (!loanResponse) history.push("/");
  return (
    <div
      className={"container"}
      dangerouslySetInnerHTML={{
        __html: `<strong>Does not qualify:</strong><br>${loanResponse &&
          loanResponse.message}<br><strong>Customer Support: 1-800-777-7777</strong>`
      }}
    />
  );
};
