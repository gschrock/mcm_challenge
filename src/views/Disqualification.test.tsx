import { shallow } from "enzyme";
import React from "react";
import { Disqualification } from "./Disqualification";

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn()
  })
}));

describe("Disqualification view", () => {
  it("renders with disqualification message", () => {
    const wrapper = shallow(
      <Disqualification
        loanResponse={{ qualified: false, message: "Not qualified message." }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
