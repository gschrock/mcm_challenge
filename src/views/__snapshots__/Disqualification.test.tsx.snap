// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disqualification view renders with disqualification message 1`] = `
<div
  className="container"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<strong>Does not qualify:</strong><br>Not qualified message.<br><strong>Customer Support: 1-800-777-7777</strong>",
    }
  }
/>
`;
